lib/brainstem/presenter_collection.rb

```diff
@@ -46,50 +49,17 @@ def presenting(name, options = {}, &block)
       # table name will be used to query the database for the filtered data
       options[:table_name] = presented_class.table_name
 
-      if searching? options
-        # Search
-        sort_name, direction = options[:primary_presenter].calculate_sort_name_and_direction options[:params]
-        scope, count, ordered_search_ids = run_search(scope, filter_includes(options).map(&:name), sort_name, direction, options)
-
-        # Load models!
-        primary_models = scope.to_a
-      else
-        # Filter
-        scope = options[:primary_presenter].apply_filters_to_scope(scope, options[:params], options)
-
-        if options[:params][:only].present?
-          # Handle Only
-          scope, count = handle_only(scope, options[:params][:only])
-        else
-          # Paginate
-          scope, count = paginate scope, options
-        end
+      options[:default_per_page] = default_per_page
+      options[:default_max_per_page] = default_max_per_page
+      options[:default_max_filter_and_search_page] = default_max_filter_and_search_page
 
-        count = count.keys.length if count.is_a?(Hash)
-
-        # Ordering
-        scope = options[:primary_presenter].apply_ordering_to_scope(scope, options[:params])
-
-        # Load models!
-        # On complex queries, MySQL can sometimes handle 'SELECT id FROM ... ORDER BY ...' much faster than
-        # 'SELECT * FROM ...', so we pluck the ids, then find those specific ids in a separate query.
-        if(ActiveRecord::Base.connection.instance_values["config"][:adapter] =~ /mysql|sqlite/i)
-          ids = scope.pluck("#{scope.table_name}.id")
-          id_lookup = {}
-          ids.each.with_index { |id, index| id_lookup[id] = index }
-          primary_models = scope.klass.where(id: id_lookup.keys).sort_by { |model| id_lookup[model.id] }
-        else
-          primary_models = scope.to_a
-        end
-      end
+      primary_models, count = strategy(options, scope).execute(scope)
```

Comment:
I am a little confused that in `FilterAndSearch` we return an arel scope, but here it would appear we want the instantiated models.